src/read_sensor_data.rs

```diff
@@ -28,6 +33,62 @@ where
     }
 }
 
+#[cfg(feature = "scaled_data")]
+impl<DI, CommE, PinE> Bmi160<DI>
+where
+    DI: ReadData<Error = Error<CommE, PinE>> + WriteData<Error = Error<CommE, PinE>>,
+{
+    /// Read latest sensor data
+    pub fn scaled_data(&mut self, selector: SensorSelector) -> Result<DataScaled, Error<CommE, PinE>> {
+        let accel_div:f32 = match self.accel_range {
+            crate::AccelerometerRange::Range2g => 16384.0f32,
+            crate::AccelerometerRange::Range4g => 8192.0f32,
+            crate::AccelerometerRange::Range8g => 4096.0f32,
+        };
+
+        let gyro_div:f32 = match self.gyro_range {
+            crate::GyroscopeRange::Range2000s => 16.4,
+            crate::GyroscopeRange::Range1000s => 32.8,
+            crate::GyroscopeRange::Range500s => 65.6,
+            crate::GyroscopeRange::Range250s => 131.2,
+            crate::GyroscopeRange::Range125s => 262.4,
+        };
+
+        let result = if selector != SensorSelector::new() {
+            let (begin, end) = get_data_addresses(selector);
+            let mut data = [0_u8; 24];
+            data[0] = begin;
+            let len = (1 + end - begin) as usize;
+            self.iface.read_data(&mut data[0..len])?;
+            let unscaled_data = get_data(selector, &data[1..], (begin - Register::MAG) as usize);
+            DataScaled {
+                accel: unscaled_data.accel.and_then(|unscaled_accel| {
+                    Some(Sensor3DDataScaled {
+                        x: unscaled_accel.x as f32 / accel_div,
+                        y: unscaled_accel.y as f32 / accel_div,
+                        z: unscaled_accel.z as f32 / accel_div,
+                    })
+                }),
+                gyro: unscaled_data.gyro.and_then(|unscaled_gyro| {
+                    Some(Sensor3DDataScaled {
+                        x: unscaled_gyro.x as f32 / gyro_div,
+                        y: unscaled_gyro.y as f32 / gyro_div,
+                        z: unscaled_gyro.z as f32 / gyro_div,
```

Comment:
Nit: It might be better implement the scaling as multiplication rather than division, as floating point division is significantly slower than multiplication on some low-end CPUs.

Reply:
Implemented in 3ab983fdde0b5bf56adcddfcfd5e64f23ba983d0 and a4cc124b330f0359cb452c8b7736521dad1bce0b, needs rebasing of course.